.github/workflows/artifact-tests.yml

```diff
@@ -0,0 +1,144 @@
+name: artifact-unit-tests
+on:
+  push:
+    paths:
+        - 'packages/artifact/**'
+  pull_request:
+    paths:
+        - 'packages/artifact/**'
+
+jobs:
+
+  build:
+    name: Build
+
+    strategy:
+      matrix:
+        runs-on: [ubuntu-latest]
+      fail-fast: false
+
+    runs-on: ${{ matrix.runs-on }}
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Set Node.js 12.x
+      uses: actions/setup-node@v1
+      with:
+        node-version: 12.x
+
+    # Need root node_modules because certain npm packages like jest are configured for the entire repository and it won't be possible
+    # without these to just compile the artifacts package
+    - name: Install root npm packages
+      run: npm install
+
+    - name: Compile artifact package
+      run: |
+        npm install
+        tsc
+      working-directory: packages/artifact
+
+    - name: Create files that will be uploaded
+      run: |
+        mkdir artifact-path
+        echo "hello" > artifact-path/world.txt
+        echo "Some large amount of text that has a compression ratio that is greater than 100%. If greater than 100%, gzip is used to upload the file" > artifact-path/gzip.txt
+
+    # Certain env variables are needed in order to make the necessary API calls. These env variables are not available in a standard shell but they
+    # are available for actions written in node. We're using a hack here to get the necessary env variables from a different context
+    - name: Get ACTIONS_RUNTIME_TOKEN env variable
+      uses: konradpabjan/Get-Actions-Env-Variable-From-Node-Context@4a3142ee3e
```

Comment:
I do not intend to ever publish this action (there is no `v1` tag or anything) so I'm just pinning it to a commit instead of branch that could regress

Reply:
Can we combine these into one call? And do we need an external action to do this?

Seems like a quick node script that we run as a local action would be more concise and checked in to this repo:


```js
# index.js
console.log(`::set-output name=ACTIONS_RUNTIME_URL::${process.env.ACTIONS_RUNTIME_URL}`)
console.log(`::set-output name= ACTIONS_RUNTIME_TOKEN::${process.env. ACTIONS_RUNTIME_TOKEN}`)
...
```

```yaml
# action.yml
runs:
  using: 'node12'
  main: 'index.js'
```

Reply:
I aggregated everything into a local action, cleans up things a little more.

Originally I was using a separate action so that it could be reused, but I don't expect this ever to be needed outside of this specific testing scenario 😂